hybrid/pkg/server/server.go

```diff
@@ -0,0 +1,272 @@
+	"github.com/spiffe/spire/pkg/server/plugin/nodeattestor/awsiid"
+	"github.com/spiffe/spire/pkg/server/plugin/nodeattestor/azuremsi"
+	nodeattestorbase "github.com/spiffe/spire/pkg/server/plugin/nodeattestor/base"
+	"github.com/spiffe/spire/pkg/server/plugin/nodeattestor/gcpiit"
+	"github.com/spiffe/spire/pkg/server/plugin/nodeattestor/k8spsat"
+)
+
+var (
+	_ pluginsdk.NeedsHostServices = (*HybridPluginServer)(nil)
+)
+
+type HybridPluginServer struct {
+	pluginList []common.Types
+	nodeattestorbase.Base
+	agentstorev1.UnimplementedAgentStoreServer
+	nodeattestorv1.UnimplementedNodeAttestorServer
+	configv1.UnsafeConfigServer
+
+	logger      hclog.Logger
+	store       agentstorev1.AgentStoreServiceClient
+	broker      pluginsdk.ServiceBroker
+	interceptor ServerInterceptorInterface
+	initStatus  error
+}
+
+func New() *HybridPluginServer {
+	interceptor := new(HybridPluginServerInterceptor)
+	return &HybridPluginServer{interceptor: interceptor}
+}
+
+func (p *HybridPluginServer) SetLogger(logger hclog.Logger) {
+	p.logger = logger
+}
+
+func (p *HybridPluginServer) BrokerHostServices(broker pluginsdk.ServiceBroker) error {
+	p.broker = broker
+
+	return nil
+}
+
+func (p *HybridPluginServer) setBrokerHostServices() error {
+	if !p.broker.BrokerClient(&p.store) {
+		return errors.New("Agent store service required")
+	}
+
+	for i := 0; i < len(p.pluginList); i++ {
+		elem := reflect.ValueOf(p.pluginList[i].Plugin)
+		method := elem.MethodByName("BrokerHostServices")
+		if method.IsValid() {
+			err := elem.MethodByName("BrokerHostServices").Call([]reflect.Value{reflect.ValueOf(p.broker)})
+			p.logger.Debug(err[0].String())
+		}
+	}
+
+	return nil
+}
+
+func (p *HybridPluginServer) Attest(stream nodeattestorv1.NodeAttestor_AttestServer) error {
+	req, err := stream.Recv()
+	if err != nil {
+		return err
+	}
+
+	p.interceptor.ResetInterceptor()
+	p.interceptor.setCustomStream(stream)
+	p.interceptor.SetContext(stream.Context())
+	p.interceptor.SetLogger(p.logger)
+	p.interceptor.SetReq(req)
+
+	payloadRequest, ok := req.Request.(*nodeattestorv1.AttestRequest_Payload)
+	if !ok {
+		return status.Errorf(codes.InvalidArgument, "request payload is required")
+	}
+
+	var messageList common.PluginMessageList = common.PluginMessageList{}
+	if err := json.Unmarshal(payloadRequest.Payload, &messageList); err != nil {
+		return status.Errorf(codes.InvalidArgument, "unable to unmarshal payload: %v", err)
+	}
+	interceptors := []ServerInterceptorInterface{}
+	for _, message := range messageList.Messages {
+		name := message.PluginName
+		processed := false
+		newReq := &nodeattestorv1.AttestRequest{
+			Request: &nodeattestorv1.AttestRequest_Payload{
+				Payload: message.PluginData,
+			},
+		}
+
+		var newInterceptor ServerInterceptorInterface = p.interceptor.SpawnInterceptor()
+		newInterceptor.SetReq(newReq)
+		interceptors = append(interceptors, newInterceptor)
+
+		for _, plugin := range p.pluginList {
+			if plugin.PluginName == name {
```

Comment:
nit: use early continue to reduce nesting
```suggestion
			if plugin.PluginName != name {
				continue
			}
```